kernel/src/mem.rs

```diff
@@ -7,8 +7,11 @@ use core::slice;
 
 use crate::callback::AppId;
 
+/// Type for specifying an AppSlice is hidden from the kernel.
 #[derive(Debug)]
 pub struct Private;
```

Comment:
@alevy is this even close to the correct text?

Reply:
It is, that’s exactly the _intended_ use of that marker type, but never used it ever in any way. So maybe we should just get rid of both of these types...